### PR TITLE
Revise Docker image build:

### PR DESCRIPTION
**Pull request by yakutovicha**
* Build a new image on every PR (with caching).
* Test the functionality of the new image on every PR.
* Build a new image on every commit to develop/master branch and every
time a new tag is created (no caching).

### CODE REVIEW

File: .github/workflows/build_and_test_docker_on_pr.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# 1. Build docker image using cached data.
+# 2. Start the docker container with local folder mounted to it.
+# 3. Check that postgresql and rabbitmq are running.
+# 4. Copy id_rsa file from the docker container to local folder.
+# 5. Restart the container and check that the id_rsa file didn't change.
+
+name: build-and-test-image-from-pull-request
+
+on:
+  [pull_request]
```
Comment (chrisjsewell) [inline .github/workflows/build_and_test_docker_on_pr.yml:13]: Don’t run on document only changes: paths-ignore: ['docs/**']

Comment (yakutovicha) [reply .github/workflows/build_and_test_docker_on_pr.yml]: done.

**Comment by yakutovicha** [comment]
An important thing to note: we need somebody to put their Dockerhub credentials in GitHub secrets: `DOCKER_USERNAME` and `DOCKER_TOKEN`. Who wants to be that somebody? Maybe @giovannipizzi?

**Comment by yakutovicha** [comment]
> An important thing to note: we need somebody to put their Dockerhub credentials in GitHub secrets: `DOCKER_USERNAME` and `DOCKER_TOKEN`. Who wants to be that somebody? Maybe @giovannipizzi?

I added my own Docker Hub username and token. Feel free to change it.

File: .github/workflows/push_image_to_dockerhub.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+name: build-and-push-to-dockerhub
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+    tags:
+      - 'v*'
```
Comment (chrisjsewell) [inline .github/workflows/push_image_to_dockerhub.yml:12]:
```suggestion
      - "v[0-9]+.[0-9]+.[0-9]+*"
```

this is what we use for the other release workflow

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
thanks @yakutovicha, just a minor nitpick

**Review by chrisjsewell — APPROVED** [review]

**Pull request merged**